lib/dpl/provider/transifex.rb

```diff
@@ -0,0 +1,45 @@
+module DPL
+  class Provider
+    class Transifex < Provider
+      experimental 'Transifex'
+
+      DEFAULT_CLIENT_VERSION = '>=0.11'
+      DEFAULT_HOSTNAME = 'https://www.transifex.com'
+
+      def install_deploy_dependencies
+        cli_version = options[:cli_version] || DEFAULT_CLIENT_VERSION
+        self.class.pip 'transifex', 'transifex', cli_version
+      end
+
+      def needs_key?
+        false
+      end
+
+      def check_auth
+        install_deploy_dependencies
+        write_transifexrc
+        context.shell 'tx status'
+      end
+
+      def push_app
+        source_push
+      end
+
+      def write_transifexrc
+        File.open(File.expand_path('~/.transifexrc'), 'w') do |f|
+          f.puts [
+            "[#{options[:hostname] || DEFAULT_HOSTNAME}]",
+            "hostname = #{options[:hostname] || DEFAULT_HOSTNAME}",
+            "username = #{options[:username]}",
+            "password = #{options[:password]}",
+            "token = #{options[:token]}",
+          ].join("\n")
+        end
+      end
+
+      def source_push
+        context.shell 'tx push --source --no-interactive', retry: true
```

Comment:
/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.31/lib/dpl/cli.rb:53:in `shell': wrong number of arguments (2 for 1) (ArgumentError)
from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.31/lib/dpl/provider/transifex.rb:41:in `source_push'

https://travis-ci.org/freenet/fred/jobs/213110817